### ISSUE TITLE
Extremely poor performance on M1 hardware with Poisson Disk Sampling enabled

### ISSUE THREAD

**Issue by nulcow** [minetest/minetest#13757]
### Minetest version

```true
Minetest 5.7.0 (OSX)
Using Irrlicht 1.9.0mt10
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="minetest.app/Contents/Resources"
STATIC_LOCALEDIR="minetest.app/Contents/Resources/locale"
```

### Active renderer

OpenGL 2.1

### Irrlicht device

Couldn't find any mention of this in Irrlicht docs, assuming Cocoa/Quartz

### Operating system and version

macOS 12.6.8 21G725 arm64

### CPU model

Apple M1 Pro (8) @ 3.2 GHz

### GPU model

Apple M1 Pro

### OpenGL version

Since OpenGL has been deprecated, I can't find my system's OpenGL version

### Summary

Whenever I turn on Poisson Disk Sampling in my graphics settings, the performance is awful. When I turn it off, I get over 10x the FPS.
I would like to play the game with dynamic shadows, but turning on the shadow filter or making the shadows higher resolution ruins the performance.
If it isn't possible to fix these performance issues, I could just use a different shadow filter or shadow method in my copy of the engine code.

### Steps to reproduce

1. Change shadow filter setting to "on" and quality to "2"
2. Change shadow resolution to 512px
3. Start Minetest Game
4. Observe poor performance

**Comment by fluxionary** [comment]
what is poisson disk sampling? i'm familiar w/ the poisson distribution. if i search using my usual search engine (https://startpage.com/) this issue shows up as the #2 result when searching for that term. none of the other top 10 results seem informative. there does seem to be a "poisson filtering (shadow_poisson_filter)" setting. is that what you mean?

**Comment by servantoftestator** [comment]
TLDR; You can't fix these performance problems without using native metal api on m1 hardware and even then it would be slow. Turning on PDS is enabling mesh shading at the opengl level, the m1 only supports vertex shaders for performance.

Highly technical answer:

There is no OpenGL driver on M1. So minetest is running through a translation layer to the native Metal driver underneath. All OpenGL rendering on M1, no matter the framework (iOS, Catalyst, macOS) has pretty terrible performance because of that.
Metal can not support mesh shaders because it requires support for geometry shaders in custom formats. The reason metal can't and won't support geometry shaders for opengl is because realitykit, the metal api for custom materials in the shader stages, supports two different types of custom metal shader functions: geometry modifiers and surface shaders. But there is no support for a geometry shader pipeline in there, and the geometry modifies vertex shaders only for realitykit's pipeline. See someone trying to implement very very slow theoretical geometry shaders on metal here https stackoverflow DOT com/questions/50557224/metal-emulate-geometry-shaders-using-compute-shaders


This issue can probably be closed as not minetest engine's problem.
